### ISSUE TITLE
Question about stage2 rl level1 training

### ISSUE THREAD

**Issue by isXinLiu** [PRIME-RL/RL-Compositionality#8]
Thanks for your work!

The training data size is 50k in the [paper](https://arxiv.org/abs/2509.25123) and [bash/section41_42/stage2_create_problems.sh](https://github.com/PRIME-RL/RL-Compositionality/blob/main/bash/section41_42/stage2_create_problems.sh), but 500k in the [huggingface](https://huggingface.co/datasets/weizechen/RL-Compositionality-Stage2-RL-Level1-TrainData). I wonder how many samples you use in this training stage.

For hyperparameters, will these changes affect the model performance greatly?
- `algorithm.filter_groups.enable=True` to `algorithm.filter_groups.enable=False`
- `actor_rollout_ref.rollout.n=16` to `actor_rollout_ref.rollout.n=8`
- `response_length=8192` to `response_length=8192`

**Comment by chenweize1998** [comment]
Hi. Regarding the data, we did generate about 500k training samples. However, since the RL run was relatively short (around 700 steps) with an effective batch size of 16, and our dynamic sampling filters out prompts with 100% or 0% accuracy, the number of unique data points actually used was closer to 40–50k. Therefore, generating 50k samples is sufficient for training and roughly matches the amount of real data we used, as we still uniformly sample and mix problems across difficulty levels. If you’d like full reproducibility, you can directly use the dataset we uploaded.

As for the hyperparameters, we haven’t done an extensive sweep yet, but feel free to experiment with them and share your results with us.

**Comment by isXinLiu** [comment]
I see. Thank you very much for your detailed explanation!
